### PR TITLE
feat: change IsActive to Status

### PR DESCRIPTION
**Pull request by Mech0z**
## Description

Instead of a boolean, status now contains the same market participant statuses as the Market Participant domain

It also changes so that GetActiveEnergySuppliersAsync and GetActiveGridAccessProvidersAsync return both Active and Passive Market Participants.

## References

* #1645

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/GreenEnergyHub.Charges.ApplyDBMigrationsApp.csproj [hunk 1]
```diff
@@ -21,6 +21,10 @@ limitations under the License.
         <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     </PropertyGroup>
 
+    <ItemGroup>
+      <None Remove="Scripts\Model\202227091553 Change Market Participant type to support all states.sql" />
+    </ItemGroup>
+
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/GreenEnergyHub.Charges.ApplyDBMigrationsApp.csproj:27]: Remove?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/Repositories/MarketParticipantRepository.cs [hunk 2]
```diff
@@ -88,7 +88,9 @@ public Task<List<MarketParticipant>> GetActiveEnergySuppliersAsync()
             return _chargesDatabaseContext
                 .MarketParticipants
                 .Where(mp => mp.BusinessProcessRole == MarketParticipantRole.EnergySupplier)
-                .Where(m => m.IsActive)
+                .Where(m =>
+                    m.Status == MarketParticipantStatus.Active ||
+                    m.Status == MarketParticipantStatus.Inactive)
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/Repositories/MarketParticipantRepository.cs:93]: Is this intended or should it be Passive?

```suggestion
                    m.Status == MarketParticipantStatus.Passive)
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/Repositories/MarketParticipantRepository.cs [hunk 3]
```diff
@@ -88,7 +88,9 @@ public Task<List<MarketParticipant>> GetActiveEnergySuppliersAsync()
             return _chargesDatabaseContext
                 .MarketParticipants
                 .Where(mp => mp.BusinessProcessRole == MarketParticipantRole.EnergySupplier)
-                .Where(m => m.IsActive)
+                .Where(m =>
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/Repositories/MarketParticipantRepository.cs:91]: Should we change the Method names for energy supplier and grid access providers so they don't say GetActive?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/Repositories/MarketParticipantRepository.cs [hunk 4]
```diff
@@ -179,7 +183,8 @@ public async Task<MarketParticipant> GetSystemOperatorOrGridAccessProviderAsync(
             return await _chargesDatabaseContext
                 .MarketParticipants
                 .SingleAsync(mp =>
-                    mp.MarketParticipantId == marketParticipantId && mp.IsActive &&
+                    mp.MarketParticipantId == marketParticipantId &&
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/Repositories/MarketParticipantRepository.cs:186]: Please edit the method comment to match

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/MarketParticipantRepositoryTests.cs [hunk 5]
```diff
@@ -52,7 +52,7 @@ public async Task AddAsync_WhenValidMarketParticipant_IsPersisted()
             var actorId = Guid.NewGuid();
             var b2CActorId = Guid.NewGuid();
             var marketParticipant = new MarketParticipant(
-                id, actorId, b2CActorId, "00001", true, MarketParticipantRole.GridAccessProvider);
+                id, actorId, b2CActorId, "00001", MarketParticipantStatus.Active, MarketParticipantRole.GridAccessProvider);
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/MarketParticipantRepositoryTests.cs:55]: General question; are we missing some test of Passive market participants?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/MarketParticipants/Handlers/MarketParticipantDomainEventMapper.cs [hunk 6]
```diff
@@ -60,5 +61,24 @@ public static class MarketParticipantDomainEventMapper
                 gridUpdatedIntegrationEvent.GridAreaId,
                 gridUpdatedIntegrationEvent.GridAreaLinkId);
         }
+
+        private static MarketParticipantStatus MapActorStatusToMarketParticipantStatus(ActorStatus status)
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/MarketParticipants/Handlers/MarketParticipantDomainEventMapper.cs:65]: Should we separate this into a Mapper class?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/GreenEnergyHub.Charges.ApplyDBMigrationsApp.csproj (hunk 1 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/GreenEnergyHub.Charges.ApplyDBMigrationsApp.csproj:27]: removed

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/Repositories/MarketParticipantRepository.cs (hunk 4 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/Repositories/MarketParticipantRepository.cs:194]: added comment on GetActiveAndPassiveGridAccessProvidersAsync and GetActiveAndPassiveEnergySuppliersAsync

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/MarketParticipantRepositoryTests.cs (hunk 5 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/MarketParticipantRepositoryTests.cs:56]: added repository test of both DDM and DDQ for all statuses

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/Repositories/MarketParticipantRepository.cs (hunk 3 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/Repositories/MarketParticipantRepository.cs]: Changed to GetActiveAndPassiveGridAccessProvidersAsync and GetActiveAndPassiveEnergySuppliersAsync

**Review by prtandrup — APPROVED** [review]

**Pull request merged**